pkg/gitops/GitOpsConfigService.go

```diff
@@ -82,17 +83,34 @@ func NewGitOpsConfigServiceImpl(Logger *zap.SugaredLogger, ciHandler pipeline.Ci
 }
 func (impl *GitOpsConfigServiceImpl) CreateGitOpsConfig(request *GitOpsConfigDto) (*GitOpsConfigDto, error) {
 	impl.logger.Debugw("gitops create request", "req", request)
+
+	existingModel, err := impl.gitOpsRepository.GetGitOpsConfigActive()
+	if err != nil && err != pg.ErrNoRows {
+		impl.logger.Errorw("error in creating new gitops config", "error", err)
+		return nil, err
+	}
+	if existingModel != nil && existingModel.Id > 0 {
+		existingModel.Active = false
+		existingModel.UpdatedOn = time.Now()
+		existingModel.UpdatedBy = request.UserId
+		err = impl.gitOpsRepository.UpdateGitOpsConfig(existingModel)
+		if err != nil {
+			impl.logger.Errorw("error in creating new gitops config", "error", err)
+			return nil, err
+		}
+	}
+
 	model := &repository.GitOpsConfig{
 		Provider:      request.Provider,
 		Username:      request.Username,
 		Token:         request.Token,
 		GitHubOrgId:   request.GitHubOrgId,
 		GitLabGroupId: request.GitLabGroupId,
 		Host:          request.Host,
```

Comment:
in this window there won't be any active config

Reply:
yes if user forcefully deactivate.

Reply:
any gitops related call will fail in this window

Reply:
> any gitops related call will fail in this window

let's make atleast one active. will add a validation for this.